addons/mail/static/src/components/message_list/message_list.xml

```diff
@@ -2,24 +2,24 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.MessageList" owl="1">
-        <div class="o_MessageList" t-att-class="{ 'o-empty': threadView and threadView.messages.length === 0 }" t-attf-class="{{ className }}" t-on-scroll="onScroll" t-ref="root">
-            <t t-if="threadView">
+        <div class="o_MessageList" t-att-class="{ 'o-empty': messageListView and messageListView.threadViewOwner and messageListView.threadViewOwner.messages.length === 0 }" t-attf-class="{{ className }}" t-on-scroll="onScroll" t-ref="root">
+            <t t-if="messageListView and messageListView.threadViewOwner">
```

Comment:
only guard on top level record necessary (other field is required and will always be set)